modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/data_renderers/ActionsDropdownRenderer.js

```diff
@@ -164,23 +167,51 @@ function ActionItem({
 
 	const onActionDropdownItemClick = context.onActionDropdownItemClick;
 
-	return (
-		<ClayDropDown.Item
-			href={link ? url : null}
-			onClick={(event) => {
-				if (onActionDropdownItemClick) {
-					onActionDropdownItemClick({
-						action,
-						event,
-						itemData,
-					});
-				}
+	const handleClick = (event) => {
+		if (onActionDropdownItemClick) {
+			onActionDropdownItemClick({
+				action,
+				event,
+				itemData,
+			});
+		}
 
-				if (!link) {
-					handleClickOnLink(event);
-				}
-			}}
-		>
+		if (!link) {
+			handleClickOnLink(event);
+		}
+	};
+
+	const getQuickActionItem = () => {
+		const commonProps = {
+			'aria-label': icon,
+			'className': 'component-action quick-action-item',
+			'title': label,
+		};
+
+		if (link) {
+			return (
+				<ClayLink {...commonProps} href={url} onClick={handleClick}>
+					<ClayIcon symbol={icon} />
+				</ClayLink>
+			);
+		}
+		else {
+			return (
+				<ClayButtonWithIcon
+					{...commonProps}
+					displayType="unstyled"
+					monospaced={false}
+					onClick={handleClick}
+					symbol={icon}
+				/>
+			);
+		}
+	};
```

Comment:
This looks very much like `LinkOrButton`. There is a [version in Clay](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/management_toolbar/ManagementToolbar.js#L18) and a [version in `frontend-taglib-clay`](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/management_toolbar/LinkOrButton.js). last time I worked on this, I avoided using `@clayui/shared` as this was internal. There is also [this closed ticket](https://github.com/liferay/clay/issues/3940). I see @tinycarol did some recent work on this. We should probably sync on this on team channel.

Reply:
Yes, it seems. I'll ask the team to know a little more about this.
However I'm not sure if I can use that component directly as I need the `ClayButtonWithIcon` component

Reply:
pushed!